### PR TITLE
Add table aws_iot_fleet_metric Closes #1994

### LINKED ISSUE #1994: Add table aws_iot_*
Add new tables (plural!!)
- `aws_iot_core_things`
- `aws_iot_core_thing_groups`
- `aws_iot_core_thing_types`
- `aws_iot_core_fleet_metrics`
- `aws_iot_greengrass_devices`
- `aws_iot_greengrass_components`
- `aws_iot_greengrass_deployments`
- `aws_iot_greengrass_devices`
- `aws_iot_greengrass_groups`

**References**
[AWS IoT Core](https://aws.amazon.com/iot-core/?nc=sn&loc=2&dn=3)
[AWS IoT Greengrass](https://aws.amazon.com/greengrass/?nc=sn&loc=2&dn=2)

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>

```
N/A
```
</details>

# Example query results
<details>
  <summary>Results</summary>

```
> select * from aws_iot_fleet_metric
+---------------------+--------------------------------------------------------------------+------------+-------------+---------------------------+---------------------------+-------------------+-----------------------+--------+--------------+---------------+------+----->
| metric_name         | arn                                                                | index_name | description | creation_date             | last_modified_date        | aggregation_field | aggregation_type_name | period | query_string | query_version | unit | vers>
+---------------------+--------------------------------------------------------------------+------------+-------------+---------------------------+---------------------------+-------------------+-----------------------+--------+--------------+---------------+------+----->
| YourFleetMetricName | arn:aws:iot:us-east-1:333333333333:fleetmetric/YourFleetMetricName | AWS_Things | <null>      | 2024-02-02T14:15:13+05:30 | 2024-02-02T14:15:13+05:30 | registry.version  | Statistics            | 60     | *            | 2017-09-30    |      | 1   >
| Metric2             | arn:aws:iot:us-east-1:333333333333:fleetmetric/Metric2             | AWS_Things | <null>      | 2024-02-02T14:15:34+05:30 | 2024-02-02T14:15:34+05:30 | registry.version  | Statistics            | 60     | *            | 2017-09-30    |      | 1   >
+---------------------+--------------------------------------------------------------------+------------+-------------+---------------------------+---------------------------+-------------------+-----------------------+--------+--------------+---------------+------+----->

```
</details>

### CODE REVIEW

**Comment by ParthaI** [comment]
The PR has a dependency on the PR: https://github.com/turbot/steampipe-plugin-aws/pull/1997, so the Golang lint is failing.

File: aws/table_aws_iot_fleet_metric.go [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+			{
+				Func: getIoTFleetMetricTags,
+				Tags: map[string]string{"service": "iot", "action": "ListTagsForResource"},
+			},
+		},
+		GetMatrixItemFunc: SupportedRegionMatrix(iotv1.EndpointsID),
+		Columns: awsRegionalColumns([]*plugin.Column{
+			{
+				Name:        "metric_name",
+				Description: "The name of the thing type.",
```
Comment (khushboo9024) [inline aws/table_aws_iot_fleet_metric.go:49]:
```suggestion
				Description: "The name of the fleet metric to describe.",
```

File: aws/table_aws_iot_fleet_metric.go [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "arn",
+				Description: "The Amazon Resource Name (ARN) for the thing type.",
```
Comment (khushboo9024) [inline aws/table_aws_iot_fleet_metric.go:54]:
```suggestion
				Description: "The ARN of the fleet metric to describe.",
```

File: aws/table_aws_iot_fleet_metric.go [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+			{
+				Name:        "unit",
+				Description: "Used to support unit transformation such as milliseconds to seconds. The unit must be supported by CW metric (https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_MetricDatum.html)",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getIoTFleetMetric,
+			},
+			{
+				Name:        "version",
+				Description: "The version of the fleet metric.",
+				Type:        proto.ColumnType_STRING,
```
Comment (khushboo9024) [inline aws/table_aws_iot_fleet_metric.go:123]: @ParthaI I believe column type should be int.

**Review by khushboo9024 — APPROVED** [review]

**Pull request merged**